File: scripts/update_html.js
const fs = require('fs');
const path = require('path');

const files = [
    "whirlpool.html", "Technology.html", "Spacer.html", 
    "Planets/Venus.html", "Planets/Uranus.html", "Planets/sun.html", 
    "Planets/Saturn.html", "Planets/Pluto.html", "Planets/Neptune.html", 
    "Planets/Mercury.html", "Planets/Mars.html", "Planets/Jupiter.html", 
    "Planets/Earth.html", "milky-way.html", "index.html", 
    "gallery.html", "forum.html", "andromeda.html", 
    "Cosmos/WhiteHole.html", "Cosmos/BlackHole.html"
];

const toggleBtnHtml = `<button id="theme-toggle" class="theme-btn" aria-label="Toggle Dark/Light Mode"></button>`;

let updated = 0;
for (let file of files) {
    let filePath = path.join("d:/projects/Cubs in spaces", file);
    if (fs.existsSync(filePath)) {
        let content = fs.readFileSync(filePath, 'utf8');
        let initialContent = content;

        // Inject button if not present
        if (!content.includes('id="theme-toggle"')) {
            content = content.replace(/(\s*)<select id="language-switcher">/, `$1${toggleBtnHtml}$1<select id="language-switcher">`);
        }
        
        // Inject script tag if not present
        let depth = file.split('/').length - 1;
        let scriptPath = depth === 0 ? "theme.js" : "../theme.js";
        
        if (!content.includes(scriptPath) && !content.includes('theme.js"')) {
            content = content.replace(/<\/body>/, `<script src="${scriptPath}"></script>\n</body>`);
        }

        if (content !== initialContent) {
            fs.writeFileSync(filePath, content, 'utf8');
            updated++;
        }
    } else {
        console.log("File not found: " + file);
    }
}
console.log(`Updated ${updated} HTML files successfully.`);

File: scripts/update_html.py
import os
import re

files = [
    "whirlpool.html", "Technology.html", "Spacer.html", 
    "Planets/Venus.html", "Planets/Uranus.html", "Planets/sun.html", 
    "Planets/Saturn.html", "Planets/Pluto.html", "Planets/Neptune.html", 
    "Planets/Mercury.html", "Planets/Mars.html", "Planets/Jupiter.html", 
    "Planets/Earth.html", "milky-way.html", "index.html", 
    "gallery.html", "forum.html", "andromeda.html", 
    "Cosmos/WhiteHole.html", "Cosmos/BlackHole.html"
]

toggleBtnHtml = '<button id="theme-toggle" class="theme-btn" aria-label="Toggle Dark/Light Mode"></button>'

base_dir = r"d:\projects\Cubs in spaces"
updated = 0

for file in files:
    file_path = os.path.join(base_dir, file.replace("/", "\\"))
    if os.path.exists(file_path):
        with open(file_path, "r", encoding="utf-8") as f:
            content = f.read()
        
        initial = content
        
        if 'id="theme-toggle"' not in content:
            # insert before language-switcher
            content = re.sub(
                r'([ \t]*)<select id="language-switcher">', 
                r'\1' + toggleBtnHtml + r'\n\1<select id="language-switcher">', 
                content
            )
        
        depth = file.count('/')
        script_path = "theme.js" if depth == 0 else "../theme.js"
        
        if 'theme.js"' not in content:
            # find last </body> and replace it
            content = re.sub(r'</body>', f'<script src="{script_path}"></script>\n</body>', content)
            
        if content != initial:
            with open(file_path, "w", encoding="utf-8") as f:
                f.write(content)
            updated += 1
    else:
        print(f"Missing: {file_path}")

print(f"Updated {updated} files")
